src/odemis/gui/util/img.py

```diff
@@ -721,70 +721,64 @@ def ar_to_export_data(projections, raw=False):
         data_dict = projection.projectAsVis()
 
         # create the image with web overlay, legend etc.
-        for pol_mode in data_dict:
-            stream_im = data_dict[pol_mode]
+        for pol_mode, stream_im in data_dict.items():
             if stream_im is None:
                 raise LookupError("Stream %s has no data selected" % (projection.name.value,))
-            plot_im = format_rgba_darray(stream_im)  # RGB -> BGRA for cairo
-            # image is always centered, fitting the whole canvas, with scale 1
-            images = set_images([(plot_im, (0, 0), (1, 1), False, None, None, None, None,
-                                  projection.name.value, None, None, {})])
-            ar_margin = int(0.2 * stream_im.shape[0])
-            ar_size = stream_im.shape[0] + ar_margin, stream_im.shape[1] + ar_margin
+
+            if stream_im.shape[0] < MIN_AR_SIZE or stream_im.shape[1] < MIN_AR_SIZE:
+                # Increase the size of the image, to fit the minimum size
+                scale = MIN_AR_SIZE / min(stream_im.shape[:2])
+                im_scale = (scale, scale)
+            else:  # Use the image as-is, and adjust the overlay/legend size to fit
+                im_scale = (1.0, 1.0)
+
+            im_size = int(round(stream_im.shape[0] * im_scale[0])), int(round(stream_im.shape[1] * im_scale[1]))
+            ar_margin = int(0.2 * im_size[0])
+            buffer_size = im_size[0] + ar_margin, im_size[1] + ar_margin
 
             # Make surface based on the maximum resolution
-            data_to_draw = numpy.zeros((ar_size[1], ar_size[0], 4), dtype=numpy.uint8)
+            data_to_draw = numpy.zeros((buffer_size[1], buffer_size[0], 4), dtype=numpy.uint8)
             surface = cairo.ImageSurface.create_for_data(
-                data_to_draw, cairo.FORMAT_ARGB32, ar_size[0], ar_size[1])
+                data_to_draw, cairo.FORMAT_ARGB32, buffer_size[0], buffer_size[1])
             ctx = cairo.Context(surface)
 
-            im = images[0]
-            buffer_center = (-ar_margin / 2, ar_margin / 2)
-            buffer_scale = (1.0, 1.0)
-            buffer_size = stream_im.shape[0], stream_im.shape[1]
-
+            # image is always centered, fitting the whole canvas, with scale 1
+            plot_im = format_rgba_darray(stream_im)  # RGB -> BGRA for cairo
+            plot_im.metadata['dc_keepalpha'] = False
             draw_image(
                 ctx,
-                im,
-                im.metadata['dc_center'],
-                buffer_center,
-                buffer_scale,
-                buffer_size,
-                1.0,
-                rotation=im.metadata['dc_rotation'],
-                shear=im.metadata['dc_shear'],
-                flip=im.metadata['dc_flip'],
-                blend_mode=im.metadata['blend_mode'],
-                interpolate_data=False
+                plot_im,
+                p_im_center=(0, 0),
+                buffer_center=(0, 0),
```

Comment:
Before, it worked because this was compensating for a bug: buffer_size was set to `stream_im.shape`, which is the size of the buffer *not* including the margin. Because the scale was always 1, it was taking a special code path which ended up making it work anyway.
Now we just indicate properly the whole buffer size, and the the buffer center and the image center are at the same place. Much easier to read, and it even works when the scale != 0.